src/game_objects/traps/arrow_trap.cpp

```diff
@@ -34,5 +34,10 @@ void ArrowTrap::setup_model() {
 }
 
 void ArrowTrap::play_trigger_animation() {
+    assert(GameState::context.client_player);
+    float distance = glm::distance(GameState::context.client_player->transform.get_position(), this->transform.get_position());
+    events::sound_effects_event(events::AudioData(AudioManager::ARROW_SWOOSH_SOUND, false, distance));
+
```

Comment:
it'd be nicer if we can move the logic for calculating distance outside of the location where the sound is played, since we will need to do this for almost all sounds. try to find a cleaner way to make it into less function calls if possible.